src/pocapi.cpp

```diff
@@ -80,3 +94,145 @@ OVMS_Status* OVMS_ServerMultiModelOptionsSetConfigPath(OVMS_ServerMultiModelOpti
+    // TODO error handling if null
+    // if (nullptr == req)
+    InferenceRequest* request = reinterpret_cast<InferenceRequest*>(req);
+    auto status = request->addParameter(parameterName, datatype, data);
+    if (!status.ok()) {
+        return reinterpret_cast<OVMS_Status*>(new Status(status));
+    }
+    return nullptr;
+}
+
+OVMS_Status* OVMS_InferenceRequestRemoveParameter(OVMS_InferenceRequest* req, const char* parameterName) {
+    // TODO error handling if null
+    // if (nullptr == req)
+    InferenceRequest* request = reinterpret_cast<InferenceRequest*>(req);
+    auto status = request->removeParameter(parameterName);
+    if (!status.ok()) {
+        return reinterpret_cast<OVMS_Status*>(new Status(status));
+    }
+    return nullptr;
+}
+
+OVMS_Status* OVMS_InferenceRequestRemoveInput(OVMS_InferenceRequest* req, const char* inputName) {
+    // TODO error handling if null
+    // if (nullptr == req)
+    InferenceRequest* request = reinterpret_cast<InferenceRequest*>(req);
+    auto status = request->removeInput(inputName);
+    if (!status.ok()) {
+        return reinterpret_cast<OVMS_Status*>(new Status(status));
+    }
+    return nullptr;
+}
+
+OVMS_Status* OVMS_InferenceRequestInputRemoveData(OVMS_InferenceRequest* req, const char* inputName) {
+    // TODO error handling if null
+    // if (nullptr == req)
+    InferenceRequest* request = reinterpret_cast<InferenceRequest*>(req);
+    auto status = request->removeInputBuffer(inputName);
+    if (!status.ok()) {
+        return reinterpret_cast<OVMS_Status*>(new Status(status));
+    }
+    return nullptr;
+}
+
+OVMS_Status* OVMS_InferenceResponseGetOutput(OVMS_InferenceResponse* res, uint32_t id, const char** name, OVMS_DataType* datatype, const uint64_t** shape, uint32_t* dimCount, void** data, size_t* bytesize, BufferType* bufferType, uint32_t* deviceId) {
+    // TODO error handling if null
+    // if (nullptr == req)
+    InferenceResponse* response = reinterpret_cast<InferenceResponse*>(res);
+    InferenceTensor* tensor = nullptr;
+    const std::string* cppName;
+    auto status = response->getOutput(id, &cppName, &tensor);
+    if (!status.ok() ||
+        (tensor == nullptr) ||
+        (cppName == nullptr)) {
+        return reinterpret_cast<OVMS_Status*>(new Status(status));
+    }
+    const Buffer* buffer = tensor->getBuffer();
+    if (nullptr == buffer) {
+        return reinterpret_cast<OVMS_Status*>(new Status(ovms::StatusCode::NOT_IMPLEMENTED));  // TODO retcode
+    }
+    *name = cppName->c_str();
+    *datatype = tensor->getDataType();
+    *shape = tensor->getShape().data();
+    *dimCount = tensor->getShape().size();
+    *bufferType = buffer->getBufferType();
+    *deviceId = buffer->getDeviceId().value_or(0);  // TODO how discriminate betwen undefined & actual device 0
+    // possibly it is not neccessary to discriminate
+    *data = const_cast<void*>(buffer->data());  // should data return const ptr?
```

Comment:
Good question. For non cpu memory it is possible that we will not be able to write to the memory directly.